internal/lua-sandbox/api/api.go

```diff
@@ -0,0 +1,575 @@
+		table.insert(lines, line)
+	end
+
+	return lines
+end
+
+function set_instruction_quota(quota)
+    local count = 0
+
+    debug.sethook(function()
+        count = count + 1
+        if count > quota then
+            error("Instruction quota exceeded (" .. quota .. " instructions).", 2)
+        end
+    end, "", 1)
+end
+
+function goLinter(path) return runFunc("goLinter", path, nil) end
+function removeFile(path) return deleteFile(path) end
+function scanDir(path, recursive) return goScanDir(path, recursive) end
+function getRootDir() return allowedDir end
+function readFileAsString(fileName) return goReadFile(fileName) end
+function getDiff() return goGetDiff() end
+function getWorkTree() return goGetWorkTree() end
+function regexMatch(text, patterns) return goRegexMatch(text, patterns) end
+function getGitObject(object) return goGetGitObject(object) end
+function getGitObjectSize(object) return goGetGitObjectSize(object) end
+function getGitObjectHash(object) return goGetGitObjectHash(object) end
+function getGitObjectPath(object) return goGetGitObjectPath(object) end
+function getGitDiff() return goGetDiff() end
+function getGitDiffFiles() return goGetWorkTree() end
+function getGitDiffOutput() return goGetDiff() end
+`
+
+func RegisterAPIFunctions(lState *lua.LState, allowedModules []string) (*lua.LState, error) {
+	lState.SetGlobal("goExecute", lState.NewFunction(goExecute))
+	lState.SetGlobal("goScanDir", lState.NewFunction(goScanDir))
+	lState.SetGlobal("allowedDir", lua.LString(sandbox.AllowedDir))
+	lState.SetGlobal("goReadFile", lState.NewFunction(goReadFile))
+	lState.SetGlobal("goRegexMatch", lState.NewFunction(goRegexMatch))
+	lState.SetGlobal("goGetDiff", lState.NewFunction(goGetDiff))
+	lState.SetGlobal("goGetWorkTree", lState.NewFunction(goGetWorkTree))
+	lState.SetGlobal("goProcessGitDiff", lState.NewFunction(goProcessGitDiff))
+	lState.SetGlobal("goCheckAddedLargeFile", lState.NewFunction(goCheckAddedLargeFile))
+	lState.SetGlobal("goCheckMergeConflict", lState.NewFunction(goCheckMergeConflict))
+	lState.SetGlobal("goCheckJSON", lState.NewFunction(goCheckJSON))
+	lState.SetGlobal("goCheckNoCommitOnBranch", lState.NewFunction(goCheckNoCommitOnBranch))
+	lState.SetGlobal("goGetGitObject", lState.NewFunction(goGetGitObject))
+	lState.SetGlobal("goGetGitObjectSize", lState.NewFunction(goGetGitObjectSize))
+	lState.SetGlobal("goGetGitObjectHash", lState.NewFunction(goGetGitObjectHash))
+	lState.SetGlobal("goGetGitObjectPath", lState.NewFunction(goGetGitObjectPath))
+
+	if err := lState.DoString(luaHelpers); err != nil {
+		return nil, err
+	}
+
+	lState.SetGlobal("hookParameters", lua.LString(""))
+	lState.SetGlobal("hookExitCode", lua.LNumber(0))
+	lState.SetGlobal("allowedExecutables", lua.LString(strings.Join(allowedModules, ",")))
+
+	return lState, nil
+}
+
+// Return all filenames in the specified directory, take a second argument to
+// specify if the scan should be recursive
+func goScanDir(l *lua.LState) int {
```

Comment:
move go implementations to a separate file?

Reply:
actually, let's have this PR be simpler and add a very basic sandbox with next to no APIs. With all of these APIs in this PR, we're not going to do justice in our review